site/en/blog/scroll-animation-performance-case-study/index.md

```diff
@@ -0,0 +1,137 @@
+
+## The scroll-driven animations CSS API versus classic JavaScript
+
+The following example progress bar is built using class JavaScript techniques.
+
+{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/gB3RmjbAlATfn9jANGLL.mp4", width="800", height="484", autoplay="true", loop="true", muted="true" %}
+
+The document responds each time the `scroll` event happens to calculate how much percentage of the `scrollHeight` the user has scrolled to.
+
+```js
+document.addEventListener("scroll", () => {
+  var winScroll = document.body.scrollTop || document.documentElement.scrollTop;
+  var height = document.documentElement.scrollHeight - document.documentElement.clientHeight;
+  var scrolled = (winScroll / height) * 100; 
+  document.getElementById("progress").style.width = scrolled + "%";
+})
+```
+
+The following demo shows the same progress bar using the new API with CSS.
+
+{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/hfvdzlpPqZEQUtIAklLx.mp4", width="800", height="484", autoplay="true", loop="true", muted="true" %}
+
+```css
+@keyframes grow-progress {
+	from {
+		transform: scaleX(0);
+	}
+	to {
+		transform: scaleX(1);
+	}
+}
+
+#progress {
+	animation: grow-progress auto linear forwards;
+	animation-timeline: scroll(block root);
+}
+```
+
+The new [animation-timeline](/articles/scroll-driven-animations/#animation-timelines) CSS feature, automatically converts a position in a scroll range into a percentage of progress, therefore doing all the heavy-lifting.
+
+Now here’s the interesting part—let’s say that you implemented a super-heavy calculation on both versions of the website that would eat up most of the main thread resources. 
+
+```js
+function someHeavyJS(){
+	let time = 0;
+	window.setInterval(function () {
+		time++;
+		for (var i = 0; i < 1e9; i++) {
+			result = i;
+		}
+		console.log(time)
+	}, 100);
+}
+```
+
+As you might have expected, the classic JavaScript version becomes janky and sluggish due to the main thread resources junction. On the other hand, the CSS version is completely unaffected by the heavy JavaScript work and can respond to the user's scroll interactions. 
+
+{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/jEhFBLHvDb9W0VFFupYz.mp4", width="800", height="485", autoplay="true", loop="true", muted="true" %}
+
+{% Codepen { user: 'nyb1030', id: 'gOBvZgR' } %}
+
+{% Codepen { user: 'nyb1030', id: 'xxyYmgx' } %}
+
+The CPU usage is completely different in DevTools, as shown in the following screenshots.
+
+{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/v6rDNCu6XO8Nh5ujx0yw.png", alt="Main thread comparison", width="800", height="450" %}
+
+The following demo shows an application of scroll driven animation created by CyberAgent. You can see that the photo fades in as you scroll.
+
+{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/xei4to5K1jmr9fCH8QFb.png", alt="CyberAgent Logo", width="800", height="178" %}
```

Comment:
This is a bit odd having a huge logo here. If you want to credit CyberAgent that's fine, but I'd suggest linking their name rather than putting in a big logo which is a bit confusing.

Reply:
I removed the image and added a link instead. Just waiting for a 👍  from Yuriko

Reply:
Got the approval from Yuriko and partner